### PR TITLE
feat(yup): added yup validation library methods

### PR DESCRIPTION
**Pull request by GoPro16**
Reopening as Draft PR.

Instead of creating our own `yup` package that we can import and use we can just create `methods` that add on functionality to the existing `yup` library and import just like a style sheet at the entry point of our projects.

```javascript
import '@availity/yup';

import * as yup from 'yup'

// use our custom methods
yup.string().dateFormat().isRequired()
```

In the case of dates the default `yup` library is using the built in `Date` lib to validate dates. However, it doesn't seem to play nice with invalid dates still as any date that is entered is presumed valid. This could be a user error and if so we can ditch the `dateFormat` method extension as a whole and default back to the `yup.date()` method for date validation.

Once we finalize the correct approach for overriding yup properties we can start looking into more validation fields such as `npi`,`zip` etc.

### CODE REVIEW

**Comment by TheSharpieOne** [comment]
I made some comments in the other PR (#124)

File: packages/yup/src/npi.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const INTEGER_REGEX = /^\d*$/;
+
+function npi(msg) {
+  return this.test({
+    name: 'isRequired',
```
Comment (nylon22) [inline packages/yup/src/npi.js]: i believe `name` should be `npi` instead of `isRequired`.

**Comment by GoPro16** [comment]
Went down a new approach rather than providing method extensions we create a new schema type `dateRange`. And then overriding the default `date` schema to use `dayjs`.

So for example a `dateRange` schema would look like:
```javascript
const schema = yup.dateRange({
// options
}).min('someStartDate').max('someEndDate');
```

File: packages/yup/src/date.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    const maxDate = dayjs(max, this.format);
+
+    return this.test({
+      message:
+        message || `Date must come before ${maxDate.format(this.format)}.`,
+      name: 'max',
+      exclusive: true,
+      params: { max },
+      test(value) {
+        return value === null || maxDate.isSameOrAfter(value);
```
Comment (nylon22) [inline packages/yup/src/date.js]: should this not be `isSameOrBefore` instead of `isSameOrAfter` ?

File: packages/yup/src/date.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    const minDate = dayjs(min, this.format);
+
+    return this.test({
+      message:
+        message || `Date must come after ${minDate.format(this.format)}.`,
+      name: 'min',
+      exclusive: true,
+      params: { min },
+      test(value) {
+        return value === null || minDate.isSameOrBefore(value, 'MM/DD/YYYY');
```
Comment (nylon22) [inline packages/yup/src/date.js]: should this not be `isSameOrAfter` instead of `isSameOrBefore` ?

File: packages/yup/src/dateRange.js [hunk 4]
```diff
@@ -8,42 +9,121 @@ dayjs.extend(customParseFormat);
...
+    this.startKey = startKey;
+    this.endKey = endKey;
+    this.format = format;
+
+    this.withMutation(() => {
+      this.transform(function mutate(value) {
+        const start = get(value, startKey);
+        const end = get(value, endKey);
+
+        if ((start && !end) || (!start && end)) {
```
Comment (nylon22) [inline packages/yup/src/dateRange.js]: what about
```
if (!start || !end) {
```

Comment (GoPro16) [reply packages/yup/src/dateRange.js]: I wish I could tell you why I did that... 🤷‍♂

**Comment by GoPro16** [comment]
I ended up replacing `dayjs` back with `moment`. The library just doesn't provide a reliable interface for custom parse formats like it should be right now. I am going to revisit `reactstrap-validation` and add it back in there as well.

**Pull request merged**